### DISCUSSION TITLE
Support for modern jenkins plugin versions

### DISCUSSION THREAD

**Discussion by NotMyFault** [renovatebot/renovate#16776] [Request Help] [answered]
### What would you like Renovate to be able to do?

Hey,

it's really cool to see that renovate is able to update Jenkins plugins from txt files. Unfortunately, the format renovate depends on is bound to a semver-like system, from what I noticed.

The Jenkins project and its plugins do not enforce semver or follow it strictly, hence a good amount of plugin authors don't use semver and rely on [CD](https://www.jenkins.io/doc/developer/publishing/releasing-cd/) versioning to publish releases.
Simply put, CD publishes a release for every PR merged, if a plugin author labels a PR as release worth.

Originally observed in https://github.com/jenkinsci/dark-theme-plugin/pull/222, where renovate attempts to update a Jenkins plugin to the last release before it started to use CD.

### If you have any ideas on how this should be implemented, please tell us here.

The CD format follows the standard `123.vabcdef456789`, an integer component determined by the Git history depth, and an informational component with the Git commit hash.
This is the syntax renovate can rely on.

Even simpler, the integer is an increasing number and will never be lower than its last release one.

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment, marked as answer]
Renovate already supports custom versioning schemes, you can do this using `packageRules` and assigning `versioning`. Something like this:

```json
{
  "packageRules": [{
    "matchPackageNames": ["dark-theme"],
    "versioning": "loose"
}
```

**Reply by NotMyFault** [reply]
Oh right, renovate does that indeed: https://github.com/jenkinsci/dark-theme-plugin/pull/229.

I was a bit surprised by the workflow chosen, considering it went from outdated -> latest before CD -> latest.

Thanks!

**Reply by viceice** [reply]
i think a better versioning is maven, it should work with those new schema
